plugins/Satellites/src/gui/SatellitesCommDialog.cpp

```diff
@@ -0,0 +1,225 @@
+			{
+				fillCommunicationsTable(satelliteID, comm.description, comm.frequency, comm.modulation);
+			}
+			adjustCommunicationsColumns();
+		}
+	}
+	// cleanup input fields
+	ui->descriptionLineEdit->setText("");
+	ui->modulationLineEdit->setText("");
+	ui->frequencySpinBox->setValue(0.0);
+}
+
+void SatellitesCommDialog::setCommunicationsHeaderNames()
+{
+	communicationsHeader.clear();
+	communicationsHeader << q_("Description");
+	communicationsHeader << q_("Frequency, MHz");
+	communicationsHeader << q_("Modulation");
+
+	ui->communicationsTreeWidget->setHeaderLabels(communicationsHeader);
+	adjustCommunicationsColumns();
+}
+
+void SatellitesCommDialog::adjustCommunicationsColumns()
+{
+	// adjust the column width
+	for (int i = 0; i < CommsCount; ++i)
+	{
+		ui->communicationsTreeWidget->resizeColumnToContents(i);
+	}
+}
+
+void SatellitesCommDialog::initListCommunications()
+{
+	ui->communicationsTreeWidget->clear();
+	ui->communicationsTreeWidget->setColumnCount(CommsCount);
+	setCommunicationsHeaderNames();
+	ui->communicationsTreeWidget->header()->setSectionsMovable(false);
+	ui->communicationsTreeWidget->header()->setDefaultAlignment(Qt::AlignCenter);
+	//ui->communicationsTreeWidget->setSelectionBehavior(QAbstractItemView::SelectRows);
+}
+
+void SatellitesCommDialog::fillCommunicationsTable(QString satID, QString description, double frequency, QString modulation)
+{
+	CommsTreeWidgetItem* treeItem = new CommsTreeWidgetItem(ui->communicationsTreeWidget);
+	treeItem->setText(CommsDescription, description);
+	treeItem->setData(CommsDescription, Qt::UserRole, satID);
+	treeItem->setText(CommsFrequency, QString::number(frequency, 'f', 3));
+	treeItem->setData(CommsFrequency, Qt::UserRole, frequency);
+	treeItem->setTextAlignment(CommsFrequency, Qt::AlignRight);
+	treeItem->setText(CommsModulation, modulation);
+}
+
+void SatellitesCommDialog::selectCurrentCommLink()
+{
+	QTreeWidgetItem* linkItem = ui->communicationsTreeWidget->currentItem();
+	ui->descriptionLineEdit->setText(linkItem->data(CommsDescription, Qt::DisplayRole).toString());
+	ui->modulationLineEdit->setText(linkItem->data(CommsModulation, Qt::DisplayRole).toString());
+	ui->frequencySpinBox->setValue(linkItem->data(CommsFrequency, Qt::UserRole).toDouble());
+}
+
+void SatellitesCommDialog::addCommData()
+{
+	QString description = ui->descriptionLineEdit->text();
+	QString modulation  = ui->modulationLineEdit->text();
+	double frequency    = ui->frequencySpinBox->value();
+	if (!description.isEmpty() && frequency>0.)
+	{
+		CommLink c;
+		c.description = description;
+		c.frequency = frequency;
+		if (!modulation.isEmpty())
+			c.modulation = modulation;
+
+		communications.append(c);
+		SatellitesMgr->getById(satelliteID)->setCommData(communications);
+		SatellitesMgr->saveCatalog();
+
+		getSatCommData();
+	}
+	else
+		QMessageBox::warning(&StelMainView::getInstance(), q_("Warning!"), q_("Please fill description and set the frequency!"), QMessageBox::Ok);
```

Comment:
I think we should give a less formal message than "Warning". Maybe:

Nothing added / Please fill in description and add frequency before adding!

Reply:
Good idea, thanks!